app.go

```diff
@@ -344,6 +346,7 @@ type Application struct {
 	ReceiptsVersion             int32
 }
 
+var checkUnsupportedBuild bool
```

Comment:
`checkUnsupportedBuild` will be shared across loomchain package. Although this should work fine since it is only used in this file, it is not a good practice as there might be other files in the future unintentionally changing or reusing this variable. So `checkUnsupportedBuild` should be a field of Application struct to make sure that only one application instance can access it.